src/parcsr_ls/par_mgr_interp.c

```diff
@@ -170,109 +187,113 @@ hypre_MGRBuildRestrict( hypre_ParCSRMatrix    *A,
 #if defined (HYPRE_USING_GPU)
       if (exec == HYPRE_EXEC_DEVICE)
       {
-         hypre_MGRBuildPDevice(A, CF_marker, num_cpts_global, restrict_type, &R);
+         hypre_MGRBuildPDevice(A, CF_marker_data, num_cpts_global, restrict_type, &RT);
       }
       else
 #endif
       {
-         hypre_MGRBuildP(A, CF_marker, num_cpts_global, restrict_type, 0, &R);
+         hypre_MGRBuildP(A, CF_marker_data, num_cpts_global, restrict_type, 0, &RT);
       }
    }
    else if (restrict_type == 1 || restrict_type == 2)
    {
 #if defined (HYPRE_USING_GPU)
       if (exec == HYPRE_EXEC_DEVICE)
       {
-         hypre_MGRBuildPDevice(AT, CF_marker, num_cpts_global, restrict_type, &R);
+         hypre_MGRBuildPDevice(AT, CF_marker_data, num_cpts_global, restrict_type, &RT);
       }
       else
 #endif
       {
-         hypre_MGRBuildP(AT, CF_marker, num_cpts_global, restrict_type, 0, &R);
+         hypre_MGRBuildP(AT, CF_marker_data, num_cpts_global, restrict_type, 0, &RT);
       }
    }
    else if (restrict_type == 3)
    {
       /* move diagonal to first entry */
       hypre_CSRMatrixReorder(hypre_ParCSRMatrixDiag(AT));
-      hypre_MGRBuildInterpApproximateInverse(AT, CF_marker, num_cpts_global, &R);
-      hypre_BoomerAMGInterpTruncation(R, trunc_factor, max_elmts);
+      hypre_MGRBuildInterpApproximateInverse(AT, CF_marker_data, num_cpts_global, &RT);
+      hypre_BoomerAMGInterpTruncation(RT, trunc_factor, max_elmts);
    }
    else if (restrict_type == 12)
    {
-      hypre_MGRBuildPBlockJacobi(AT, A_FFT, A_FCT, NULL, blk_size, CF_marker, num_cpts_global,
-                                 &R);
+      hypre_MGRBuildPBlockJacobi(AT, A_FFT, A_FCT, NULL, blk_size, CF_marker_data, &RT);
    }
    else if (restrict_type == 13) // CPR-like restriction operator
```

Comment:
Does this new implementation require A_CF to be present always? Since it is only needed for this restrict_type, it may be NULL, right? If so, should we check here to make sure it is not NULL?

Reply:
Yes, good idea, I'll add the check